lib/iron_mq/message.rb

```diff
@@ -0,0 +1,92 @@
+require 'iron_mq/entity'
+
+module IronMQ
+  class Message
+    include Entity
+
+    define_fields :id, :body, :reservation_id, :reserved_count,
+                  :subscribers, :headers, :push_statuses
+
+    def initialize(client, queue_name, msg_fields)
+      @client = client
+      if queue_name.nil? || queue_name.empty?
+        fail ArgumentError, 'queue_name must be non-empty String'
+      end
+      @queue_name = queue_name
+      self.entity = msg_fields
+
+      fail ArgumentError, 'msg_fields must contain message ID' unless id
+    end
+
+    def get!
```

Comment:
Updated. Added non-bang `get` and `get_push_statuses`.
In real, about all the methods in the library are dangerous. They:
1. Can raise exception
2. Changes the state of the entity on service's side
3. Changes receiver

I chose to use `!` to only on methods, that changes receiver.